src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1517,55 +1555,28 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
             impl_candidate = impl_types::ocl;
         }
 
-        for (auto& fo : node.get_fused_primitives()) {
-            if (fo.node->is_type<eltwise>()) {
-                // FC checkings
-                if (node.is_type<fully_connected>()) {
-                    auto in_layout = node.get_dependency(fo.dep_start_idx).get_output_layout();
-                    auto out_layout = node.get_output_layout();
-                    auto in_dt = in_layout.data_type;
-                    auto out_dt = out_layout.data_type;
-                    // if it is not eltwise sum and input is full tensor
-                    if ((out_layout.count() == in_layout.count()) && in_dt != out_dt
-                        && (data_type_traits::is_floating_point(in_dt) || data_type_traits::is_floating_point(out_dt))
-                        && onednn_add_fusing_helpers::is_full_tensor(in_layout)) {
-                        impl_candidate = impl_types::ocl;
-                        break;
-                    }
+        if (node.is_type<fully_connected>()) {
+            if (!is_node_for_onednn(node, node.as<fully_connected>()))
+                impl_candidate = impl_types::ocl;
 
-                    // WA: onednn sum/binary_add post-op are not supported due to perf drop.
-                    auto add_type = onednn_add_fusing_helpers::get_add_fusing_type(node, fo);
-                    if (add_type == add_fusing_type::sum || add_type == add_fusing_type::binary_per_tensor || add_type == add_fusing_type::binary_per_oc) {
-                        impl_candidate = impl_types::ocl;
-                        break;
-                    }
-                // Gemm checkings
-                // TODO: investigate why currently onednn gemm has some "sum" post-op restrictions
-                // which don't correlate with fc checkings in the code above
-                // Temprorary WA: disable onednn gemm with sum post-op inside
-                } else {
+            // WA : Use cldnn FC due to perf drop of small batch size until onednn FC improve perf
+            if (node.get_output_layout().batch() < 32)
+                impl_candidate = impl_types::ocl;
```

Comment:
I think this part is the essence. Could you explain why you changed other parts?

Reply:
From a short meeting, we decided to add this func is_mode_for_onednn.
Just remove unnecessary comment about dep_start_idx.